source/agora/common/BlockStorage.d

```diff
@@ -301,8 +280,17 @@ public class BlockStorage
         if (finds.empty)
             return false;
 
-        this.readBlockAtPosition(block, finds.front.position);
-        return true;
+        bool success = true;
```

Comment:
In general, it is much better to assign failure by default, not success.

However there is no need for this boolean. This can be rewritten to:

```D
try
{
    this.readBlockAtPosition(block, finds.front.position);
    return true;
}
catch (Exception ex)
{
    this.writeLog("BlockStorage.readBlock: ", ex);
    return false;
}
```